src/sentry/relay/config/metric_extraction.py

```diff
@@ -144,35 +145,12 @@ def _get_alert_metric_specs(
                 tags={"prefilling": prefilling, "dataset": alert_snuba_query.dataset},
             )
 
-            if result := _convert_snuba_query_to_metric(
-                project, alert_snuba_query, prefilling, use_updated_env_logic=True
-            ):
-                _log_on_demand_metric_spec(
-                    project_id=project.id,
-                    spec_for="alert",
-                    spec=result,
-                    id=alert.id,
-                    field=alert_snuba_query.aggregate,
-                    query=alert_snuba_query.query,
-                    prefilling=prefilling,
-                )
-                metrics.incr(
-                    "on_demand_metrics.on_demand_spec.for_alert",
-                    tags={"prefilling": prefilling},
-                )
-                specs.append(result)
-
-            # In case the query has an environment, we want to extract with the old environment logic, since we found
-            # a bug in the old logic and this requires us to extract the same metric in parallel but with a different
-            # query hash.
-            if alert_snuba_query.environment_id is not None:
-                if result := _convert_snuba_query_to_metric(
-                    project, alert_snuba_query, prefilling, use_updated_env_logic=False
-                ):
+            if results := _convert_snuba_query_to_metric(project, alert_snuba_query, prefilling):
```

Comment:
Maybe we should make it plural like 
```suggestion
            if results := _convert_snuba_query_to_metrics(project, alert_snuba_query, prefilling):
```

I am not so sure though, since it may confuse people.